ufo2fontir/src/source.rs

```diff
@@ -845,6 +863,79 @@ impl Work<Context, WorkError> for FeatureWork {
     }
 }
 
+impl Work<Context, WorkError> for KerningWork {
+    fn exec(&self, context: &Context) -> Result<(), WorkError> {
+        debug!("Kerning for {:#?}", self.designspace_file);
+
+        let designspace_dir = self.designspace_file.parent().unwrap();
+        let static_metadata = context.get_final_static_metadata();
+        let master_locations = master_locations(&static_metadata.axes, &self.designspace.sources);
+
+        let mut kerning = Kerning::default();
+        for source in self.designspace.sources.iter() {
+            let pos = master_locations.get(&source.name).unwrap();
+            let ufo_dir = designspace_dir.join(&source.filename);
+            let data_request = norad::DataRequest::none().groups(true).kerning(true);
+            let font = norad::Font::load_requested_data(&ufo_dir, data_request)
+                .map_err(|e| WorkError::ParseError(ufo_dir, format!("{e}")))?;
+
+            kerning.groups = font
+                .groups
```

Comment:
note that UFO's groups.plist is not just for _kerning_ groups, technically there can be any named list of glyphs and the use depend on the authoring tool. The kerning groups proper are prefixed with "public.kern1." and "public.kern2.", you should ignore anything else

Reply:
TIL! ty, will fix